### ISSUE TITLE
cannot solve pyserial dependencies in 3.10 on Windows

### ISSUE THREAD

**Issue by p-elsie** [pyinstaller/pyinstaller#6947]
It works flawlessly in Python 2.7, using pyinstaller==3.6. 

In Python 3.8 or 3.10, I have issues, and I can't solve the dependency puzzle. 

I was trying to compile this: 

https://github.com/p-elsie/ttyUtil

**Comment by bwoodsend** [comment]
You're probably just mixing up your Python environments then. It works fine for me.

**Comment by p-elsie** [comment]
@bwoodsend did you actually test it on a serial port? Because that's the real test. There was this issue ages ago:

https://github.com/pyinstaller/pyinstaller/issues/1019

On that issue, @matysec observed that if you got it right, it included these files (when you compiled to folder):

msvcm90.dll  
msvcr90.dll
msvcp90.dll

But otherwise, you did not. And when you go to use a serial port, it reported that it could not satisfy "import serial". 

My experience was just like that.

**Comment by p-elsie** [comment]
Here's my result (after uninstalling and reinstalling Python 3.10.5):

ttyUtil.exe  --port=COM4 --string="1 123 32"  -v
Traceback (most recent call last):
  File "ttyUtil.py", line 60, in <module>
    main(sys.argv[1:])
  File "ttyUtil.py", line 36, in main
    ser = serial.Serial(port, int(baud), timeout=1)
  File "site-packages\serial\serialwin32.py", line 33, in __init__
  File "site-packages\serial\serialutil.py", line 244, in __init__
  File "site-packages\serial\serialwin32.py", line 64, in open
serial.serialutil.SerialException: could not open port 'COM4': FileNotFoundError(2, 'The system cannot find the file specified.', None, 2)
[980] Failed to execute script ttyUtil

C:\Users\Patrick\pctemp>dist\ttyUtil.exe  --port=COM4 --string="1 123 32"  -v
Traceback (most recent call last):
  File "ttyUtil.py", line 60, in <module>
    main(sys.argv[1:])
  File "ttyUtil.py", line 36, in main
    ser = serial.Serial(port, int(baud), timeout=1)
  File "serial\serialwin32.py", line 33, in __init__
  File "serial\serialutil.py", line 244, in __init__
  File "serial\serialwin32.py", line 64, in open
serial.serialutil.SerialException: could not open port 'COM4': FileNotFoundError(2, 'The system cannot find the file specified.', None, 2)
[3164] Failed to execute script 'ttyUtil' due to unhandled exception!

And if I compile to folder, these are the files in it:

base_library.zip
libcrypto-1_1.dll
libffi-7.dll
python310.dll
select.pyd
ttyUtil.exe
unicodedata.pyd
VCRUNTIME140.dll
_bz2.pyd
_ctypes.pyd
_decimal.pyd
_hashlib.pyd
_lzma.pyd
_socket.pyd

Compare that, to what I get when I do the same with Python 2.7 and pyinstaller 3.6:

bz2.pyd
Microsoft.VC90.CRT.manifest
msvcm90.dll
msvcp90.dll
msvcr90.dll
python27.dll
select.pyd
ttyUtil.exe
ttyUtil.exe.manifest
unicodedata.pyd
_ctypes.pyd
_hashlib.pyd

It looks like a regression of https://github.com/pyinstaller/pyinstaller/issues/1019 to me.

**Comment by p-elsie** [comment]
Oops - I moved the USB device that was on COM4 to another host, where I am using the compiled binary. 

With COM1 it works fine (built-in COM port, nothing is actually connected). So, I guess it was a environment issue after all. Thanks. 

Result: 

ttyUtil.exe --port=COM1 --string="hello world"  -v
port is COM1
baud is 9600
string is "hello world"
response is ""

I guess it was due to the PATH variable (in the cmd console that I still have open), which still looks like this:

PATH=C:\Program Files (x86)\Intel\iCLS Client\;C:\Program Files\Intel\iCLS Client\;C:\Python38-32\Scripts\;C:\Python38-32\;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;...

Before I reinstalled Python 3.10.5, I uninstalled 3.8 and deleted C:\Python38-32. Since it's gone, pyinstaller moves on to looking in the folders adjacent to where it, pyinstaller.exe is - or so I assume. And that's why it's working correctly now.
